pkg/feeds/factory/factory.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 The Swarm Authors. All rights reserved.
```

Comment:
TODO: This needs to be implemented directly on the feed struct which should have an index type field.

Reply:
but then how do you inject the mock factory with the mock update?